### ISSUE TITLE
opam build failure due to missing oclock dependency

### ISSUE THREAD

**Issue by avsm** [xapi-project/xenopsd#161]
```
#=== ERROR while installing xapi-xenopsd.0.9.44 ===============================#
# opam-version         1.2.0 (75c4bf0dd32913f9218ffa882e683e01d6cdfb0f)
# os                   linux
# command              make
# path                 /home/opam/.opam/4.01.0/build/xapi-xenopsd.0.9.44
# compiler             4.01.0
# exit-code            2
# env-file             /home/opam/.opam/4.01.0/build/xapi-xenopsd.0.9.44/xapi-xenopsd-17557-ab22a1.env
# stdout-file          /home/opam/.opam/4.01.0/build/xapi-xenopsd.0.9.44/xapi-xenopsd-17557-ab22a1.out
# stderr-file          /home/opam/.opam/4.01.0/build/xapi-xenopsd.0.9.44/xapi-xenopsd-17557-ab22a1.err
### stdout ###
# ...[truncated]
# /home/opam/.opam/4.01.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -thread -I xc -I lib -o xc/hotplug.cmo xc/hotplug.ml
# /home/opam/.opam/4.01.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -thread -I xc -I lib -o xc/memory.cmo xc/memory.ml
# /home/opam/.opam/4.01.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -thread -I xc -I lib -o xc/stubdom.cmi xc/stubdom.mli
# /home/opam/.opam/4.01.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -thread -I xc -I lib -o xc/xenctrlext.cmi xc/xenctrlext.mli
# /home/opam/.opam/4.01.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -thread -I xc -I lib -o xc/xenstore_watch.cmo xc/xenstore_watch.ml
# /home/opam/.opam/4.01.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -syntax camlp4o -thread -I xc -I lib -o xc/xenops_server_xen.cmo xc/xenops_server_xen.ml
# + /home/opam/.opam/4.01.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -syntax camlp4o -thread -I xc -I lib -o xc/xenops_server_xen.cmo xc/xenops_server_xen.ml
# File "xc/xenops_server_xen.ml", line 165, characters 28-42:
# Error: Unbound module Oclock
# Command exited with code 2.
```

**Comment by simonjbeaumont** [comment]
Weird; it's in the `_oasis` file:
https://github.com/xapi-project/xenopsd/blob/master/_oasis#L33

and the `opam` file:
https://github.com/xapi-project/xenopsd/blob/master/opam#L29

**Comment by djs55** [comment]
Is it missing from the opam-repository opam file? It'll be a snapshot of
the one in the repo when the release was tagged.

On Tuesday, January 13, 2015, Si Beaumont <notifications@github.com> wrote:

> Weird; it's in the _oasis file:
> https://github.com/xapi-project/xenopsd/blob/master/_oasis#L33
>
> and the opam file:
> https://github.com/xapi-project/xenopsd/blob/master/opam#L29
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xapi-project/xenopsd/issues/161#issuecomment-69715702>
> .
>


-- 
Dave Scott

**Comment by avsm** [comment]
Here's a full log of the xenopds build failure. 

http://www.recoil.org/~avsm/opam-bulk/logs/local-ubuntu-14.04-ocaml-4.02.1/raw/xapi-xenopsd.html

**Comment by djs55** [comment]
Thanks, will take a look later!

**Comment by djs55** [comment]
oclock is definitely in the opam file in the opam-repo and opam builds it:
```
[oclock: make install] Command started
+ make "install" (CWD=/home/opam/.opam/system/build/oclock.0.4.0)
- ocamlfind install oclock oclock.cma oclock.cmxa liboclock.a oclock.cmi oclock.a META -dll dlloclock.so 
- Installed /home/opam/.opam/system/lib/oclock/oclock.a
- Installed /home/opam/.opam/system/lib/oclock/oclock.cmi
- Installed /home/opam/.opam/system/lib/oclock/liboclock.a
- Installed /home/opam/.opam/system/lib/oclock/oclock.cmxa
- Installed /home/opam/.opam/system/lib/oclock/oclock.cma
- Installed /home/opam/.opam/system/lib/oclock/META
- Installed /home/opam/.opam/system/lib/stublibs/dlloclock.so
- Installed /home/opam/.opam/system/lib/stublibs/dlloclock.so.owner
```
but it's not given as a `-package` to ocamlfind which implies an _oasis problem:
```
- /home/opam/.opam/system/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -syntax camlp4o -thread -I xc -I lib -o xc/xenops_server_xen.cmo xc/xenops_server_xen.ml
- + /home/opam/.opam/system/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -package cohttp -package fd-send-recv -package forkexec -package rpclib -syntax camlp4o -package rpclib.syntax -package sexplib -syntax camlp4o -package sexplib.syntax -package threads -package threads.posix -package uri -package uuidm -package uutf -package xcp -package xcp-inventory -package xcp-rrd -package xcp.memory -package xcp.rrd -package xcp.storage -package xcp.xen -package xenctrl -package xenstore -package xenstore.unix -package xenstore_transport -package xenstore_transport.unix -package xmlm -syntax camlp4o -thread -I xc -I lib -o xc/xenops_server_xen.cmo xc/xenops_server_xen.ml
- File "xc/xenops_server_xen.ml", line 165, characters 28-42:
- Error: Unbound module Oclock
```

**Comment by djs55** [comment]
... easy enough to repro in an Ubuntu 14.04 VM

**Comment by djs55** [comment]
Ah, this was in fact the only change in v0.9.45 -- I'll replace the v0.9.44 package with a v0.9.45 version.

**Comment by djs55** [comment]
Oh actually v0.9.45 brings in further dependencies. I'll release a fixed 0.9.44.1

**Comment by djs55** [comment]
OK, fixed and released.
